lib/recipe.rb

```diff
@@ -0,0 +1,45 @@
+class Recipe
+  attr_accessor  :label, :image, :source, :uri, :healthLabels, :url, :ingredientLines
+  def initialize(label, image, source, uri, healthLabels, url, ingredientLines )
+
+    raise_error(label)
+    raise_error(image)
+    raise_error(source)
+    raise_error(uri)
+    raise_error(url)
+    raise_error(ingredientLines)
```

Comment:
I would say you're over-validating here. In general you should only validate fields that will cause your app to break if they're not there. In this case that's `label`, `uri` and possibly `image`. Anything beyond that is an unnecessary dependency on the data coming from the API, and is an extra thing for you to test.